### PR TITLE
fix: add topology.disk.csi.azure.com/zone to normalized labels

### PR DESCRIPTION
**Pull request by Bryce-Soghigian**
fix: adding topology label to normalized labels

Fixes # 

**Description**
workload using topology.disk-csi-azure.com/zone in node affinity would fail to schedule. Karpenter has a concept for translating these labels in its scheduling code to normalize the values. This PR Changes topology.disk.csi.azure.com to map to the standard topology label. 

This PR also adds additional testing around well known labels and pod node selector requests. This discovered the unimplemented Isolated size and Confidential well known labels we expose. Removing them until we have more concrete usecases for them. They are not in the docs so no docs update required. 

**How was this change tested?**
Scale up, workload with `topology.disk.csi.azure.com/zone1 schedules fine. 

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note

```

### CODE REVIEW

**Review by tallaxes — CHANGES_REQUESTED** [review]
The fix is correct, but please add corresponding tests

File: pkg/providers/instancetype/suite_test.go [hunk 1]
```diff
@@ -813,6 +813,60 @@ var _ = Describe("InstanceType Provider", func() {
...
+				v1.LabelTopologyZone:             "test-zone-1a",
+				v1.LabelInstanceTypeStable:       "Standard_NC24ads_A100_v4",
+				v1.LabelOSStable:                 "linux",
+				v1.LabelArchStable:               "amd64",
+				corev1beta1.CapacityTypeLabelKey: "on-demand",
+				// Well Known to AKS
+				v1alpha2.LabelSKUName:                      "Standard_NC24ads_A100_v4",
+				v1alpha2.LabelSKUFamily:                    "N",
+				v1alpha2.LabelSKUVersion:                   "4",
+				v1alpha2.LabelSKUConfidential:              "false",
```
Comment (Bryce-Soghigian) [inline pkg/providers/instancetype/suite_test.go:832]: @tallaxes  https://github.com/search?type=code&q=LabelSKUConfidential we don't add this label  or isolatedSize to the instancetype requirements so these tests won't work with those labels. Thoughts on removing them? I don't see them used anywhere

Comment (tallaxes) [reply pkg/providers/instancetype/suite_test.go:832]: Corrent, neither selectors are supported yet (and they are not in the public docs), should not be in WellKnowLabels. Cleanup was on the TODO list, but great to have a test that catches this! 👍

Comment (Bryce-Soghigian) [reply pkg/providers/instancetype/suite_test.go]: Removed them in this PR.

File: pkg/providers/instancetype/suite_test.go [hunk 2]
```diff
@@ -813,6 +813,58 @@ var _ = Describe("InstanceType Provider", func() {
...
+				v1alpha2.LabelSKUEncryptionAtHostSupported: "true",
+				v1alpha2.LabelSKUStoragePremiumCapable:     "true",
+				v1alpha2.AKSLabelCluster:                   "test-cluster",
+				v1alpha2.LabelSKUGPUName:                   "A100",
+				v1alpha2.LabelSKUGPUManufacturer:           "nvidia",
+				v1alpha2.LabelSKUGPUCount:                  "1",
+				v1alpha2.LabelSKUCPU:                       "24",
+				v1alpha2.LabelSKUMemory:                    "8192",
+				v1alpha2.LabelSKUAccelerator:               "A100",
+
```
Comment (tallaxes) [inline pkg/providers/instancetype/suite_test.go:845]: nit: remove empty line

File: pkg/providers/instancetype/suite_test.go [hunk 3]
```diff
@@ -813,6 +813,58 @@ var _ = Describe("InstanceType Provider", func() {
...
+				v1alpha2.LabelSKUName:                      "Standard_NC24ads_A100_v4",
+				v1alpha2.LabelSKUFamily:                    "N",
+				v1alpha2.LabelSKUVersion:                   "4",
+				v1alpha2.LabelSKUStorageCacheSize:          "0", // Either Temp disk or cache disk can be empty :)
+				v1alpha2.LabelSKUStorageTempMaxSize:        "0",
+				v1alpha2.LabelSKUStorageEphemeralOSMaxSize: "53.6870912",
+				v1alpha2.LabelSKUAcceleratedNetworking:     "true",
+				v1alpha2.LabelSKUEncryptionAtHostSupported: "true",
+				v1alpha2.LabelSKUStoragePremiumCapable:     "true",
+				v1alpha2.AKSLabelCluster:                   "test-cluster",
```
Comment (tallaxes) [inline pkg/providers/instancetype/suite_test.go:838]: This one is a bit special; we need a better way to handle it. For now maybe just put it last (or in a separate category?) so it does not get lost ...

**Review by tallaxes — APPROVED** [review]
/test

**Comment by tallaxes** [comment]
Passing tests: https://github.com/Azure/karpenter-provider-azure/actions/runs/8133783595

**Review by tallaxes — APPROVED** [review]

**Pull request merged**